Significant refactor of the GCP Terraform code for deploying validators on mainnet

alchemydc opened the pull request:
### Description
* updated to support RC1/mainnet
* bootstrap.sh makes configuring a separate GCP project for Terraformed Celo infrastructure easy
* Adds Stackdriver support for logging, monitoring and alerting.
* Includes Celo/Geth specific Stackdriver Monitoring Dashboard
<img width="1371" alt="Screen Shot 2020-05-23 at 22 13 32" src="https://user-images.githubusercontent.com/1643352/82745526-b0553d00-9d42-11ea-9360-4c261145a2cd.png">

* Fixes Docker start/stop via systemd to avoid chaindata corruption
* Adds support for backup/restore of chaindata via cloud storage to avoid slow p2p sync
* Rotates geth logs (/var/log/daemon.log) to prevent boot disk from filling up
* Adds support for a public static website bucket for hosting metadata, etc.
* Fixes security issues associated with txnode exposing RPC to the Internet
* Moves secrets out of git, and into terraform.tfvars (per custom)
* Fixes to support new mainnet attestation service docker image
* Fixes stats support on proxy node (by unlocking an otherwise unused ethereum account on proxy)

### Tested

Used to successfully run infrastructure on RC1 / mainnet

alchemydc committed:
Add support for unlocking proxy address on proxy node, so proxy will correctly report to stats server.

alchemydc committed:
Merge branch 'master' into master

alchemydc committed:
proxy can have >1 account address

alchemydc committed:
Merge branch 'master' of github.com:alchemydc/celo-monorepo

alchemydc committed:
Merge remote-tracking branch 'upstream/master'

alchemydc committed:
Move secrets out of git, add stackdriver support for logging and monitoring, and lots of other improvements

jcortejoso commented:
@alchemydc Awesome! Thanks for contributing.

jcortejoso commented on packages/terraform-modules-public/gcp/example/main.tf:
```diff
@@ -115,3 +116,110 @@ module "celo_cluster" {
+  ]
+}
+
+#resource "google_storage_bucket" "public_www_bucket" {
+#  name = var.public_www_fqdn
+#  location = "US"
+#  force_destroy = true
+#
+#  website {
+#    main_page_suffix = "index.html"
+#    not_found_page   = "404.html"
+#  }
+#  cors {
+#    origin          = ["https://${var.public_www_fqdn}"]
+#    method          = ["GET", "HEAD"]
+#    response_header = ["*"]
+#    max_age_seconds = 3600
+#  }
+#}
+
+#resource "google_storage_bucket_iam_binding" "public_www_binding_read" {
+#  bucket = var.public_www_fqdn
+#  role = "roles/storage.objectViewer"
+#  members = [
+#    "allUsers"
+#  ]
+#}
```
Should this be removed?

alchemydc replied:
No.  Validators need to host metadata on a publicly accessible web server, so I enabled this to make it easy to do so on GCS.  Agree it was confusing having it in there but commented so, so it's now enabled.

https://docs.celo.org/celo-codebase/protocol/identity/metadata

jcortejoso reviewed (approved):
Great improvement from current setup!

jcortejoso committed:
Merge branch 'master' into master

alchemydc committed:
Enable public static GCS bucket for metadata

alchemydc committed:
Merge branch 'master' of github.com:alchemydc/celo-monorepo

alchemydc committed:
Merge branch 'master' of github.com:celo-org/celo-monorepo

alchemydc committed:
Merge branch 'master' of github.com:celo-org/celo-monorepo

alchemydc committed:
Merge branch 'master' of github.com:celo-org/celo-monorepo

alchemydc committed:
update to debian10, use rsync for chaindata backup/restore, tune rsyslog, remove compilers

alchemydc committed:
create GCS bucket for rsynced chaindata backup and restore. disable GCS for terraform state by default (for security). update docs

alchemydc commented:
@jcortejoso note that we've updated to use debian10, switched to using rsync for backup and restore of chaindata, and made some security improvements.  Hoping to get this merged sooner or later :)

alchemydc committed:
start geth for slow sync from p2p network if no chaindata backup is available

alchemydc committed:
start geth for slow p2p sync from genesis if warp restore fails

alchemydc committed:
bump attestation docker image to mainnet tag

alchemydc committed:
bugfix hardcoded project name

alchemydc committed:
bump geth to fix critical security issue, massively improve docs, tune logging

alchemydc committed:
Merge branch 'master' of github.com:celo-org/celo-monorepo

alchemydc committed:
bump celo docker image to mainnet tag, which is v1.0.1 as of now and includes security fix

alchemydc committed:
properly hup rsyslogd after logrotate. reduce swap to 1G on txnode, automatically backup chaindata via rsync to GCS nightly and tarball to GCS weekly

alchemydc committed:
run fast rsync backup nightly, slow tarball backup weekly

Commit:
Merge branch 'master' into master

mergify[bot] committed:
Merge branch 'master' into master
